src/openforms/prefill/contrib/family_members/filters.py

```diff
@@ -0,0 +1,133 @@
+from datetime import datetime
+from typing import Literal
+
+from openforms.contrib.haal_centraal.clients import get_brp_client
+from openforms.contrib.haal_centraal.clients.brp import NaturalPersonDetails
+from openforms.contrib.haal_centraal.constants import BRPVersions
+from openforms.utils.date import get_today
+
+from .constants import HC_DECEASED_ATTRIBUTES
+
+
+def exclude_deceased_for_haal_centraal(
+    results: list[NaturalPersonDetails],
+    hc_version: Literal[BRPVersions.v13, BRPVersions.v20],
+) -> list[NaturalPersonDetails]:
+    bsn_numbers = [member["bsn"] for member in results]
+
+    with get_brp_client() as client:
+        # v2 of HaalCentraal supports request for multiple bsn numbers
+        if hc_version == BRPVersions.v20:
+            members = client.find_person(
+                bsn_numbers, attributes=HC_DECEASED_ATTRIBUTES[hc_version]
+            )
+        # v1 of HaalCentraal requires separate request for each bsn
+        else:
+            members = []
+            for bsn in bsn_numbers:
+                member = client.find_person(
+                    bsn, attributes=HC_DECEASED_ATTRIBUTES[hc_version]
+                )
+                if member:
+                    members.append(member)
+
+        if not members:
+            return results
+
+        deceased_bsns = [
+            member["burgerservicenummer"]
+            for member in members
+            if isinstance(member, dict) and "overlijden" in member
+        ]
+        eligible_members = [
+            result for result in results if result["bsn"] not in deceased_bsns
+        ]
+        return eligible_members
+
+
+def is_eligible_based_on_age(
+    provided_age: int | None = None,
+    date_of_birth: str | None = None,
+    date_of_birth_precision: Literal["date", "year_month", "year"] | None = None,
```

Comment:
I'd really define a helper `type` for this `Literal` so that you don't have to repeat it everywhere :)